src/python/pants/task/simple_codegen_task.py

```diff
@@ -246,26 +268,43 @@ def _inject_synthetic_target(
     """Create, inject, and return a synthetic target for the given target and workdir.
 
     :param target: The target to inject a synthetic target for.
-    :param target_workdir: The work directory containing the generated code for the target.
+    :param target_workdir: The base directory for the new synthetic target.
     :param fingerprint: The fingerprint to create the synthetic target
-           with to avoid re-fingerprinting
-    :param mark_transitive_invalidation_hash_dirty: Whether to walk the build graph to invalidate
-           dependees of the target.
+           with to avoid re-fingerprinting.
     """
+
+    synthetic_target_type = self.synthetic_target_type(target)
+    target_workdir = self.synthetic_target_dir(target, target_workdir)
+    synthetic_extra_dependencies = self.synthetic_target_extra_dependencies(target, target_workdir)
+
     copied_attributes = {}
     for attribute in self._copy_target_attributes:
       copied_attributes[attribute] = getattr(target, attribute)
 
-    target_workdir = self.synthetic_target_dir(target, target_workdir)
+    if self._supports_exports(synthetic_target_type):
+      extra_exports = self.synthetic_target_extra_exports(target, target_workdir)
+
+      extra_exports_not_in_extra_dependencies = set(extra_exports).difference(
+        set(synthetic_extra_dependencies))
+      if len(extra_exports_not_in_extra_dependencies) > 0:
```

Comment:
Alright, I understand what's going on here now, and I would be fine with not validating this here, and letting it be validated later.

Reply:
I still would prefer to have this here because it's a more specific error.